pkg/neorpc/filters.go

```diff
@@ -126,14 +147,76 @@ func (f *NotificationFilter) Copy() *NotificationFilter {
 		res.Name = new(string)
 		*res.Name = *f.Name
 	}
+	if len(f.Parameters) != 0 {
+		res.Parameters = slices.Clone(f.Parameters)
+	}
+	f.parametersCache = nil
 	return res
 }
 
+// ParametersSI returns [stackitem.Item] version of [NotificationFilter.Parameters]
+// according to [smartcontract.Parameter.ToStackItem]; Notice that the result is
+// cached internally in [NotificationFilter] for efficiency, so once you call this
+// method it will not change even if you change any structure fields. If you need
+// to update parameters, use [NotificationFilter.Copy].
+// It mainly should be used by server code. Must not be used concurrently.
+func (f *NotificationFilter) ParametersSI() ([]stackitem.Item, error) {
+	if len(f.Parameters) == 0 {
+		return nil, nil
+	}
+	if f.parametersCache != nil {
+		return f.parametersCache, nil
+	}
+	f.parametersCache = make([]stackitem.Item, 0, len(f.Parameters))
+	for i, p := range f.Parameters {
+		si, err := p.ToStackItem()
+		if err != nil {
+			f.parametersCache = nil
+			return nil, fmt.Errorf("converting %d parameter to stack item: %w", i, err)
+		}
+		f.parametersCache = append(f.parametersCache, si)
+	}
+	return f.parametersCache, nil
+}
+
+// MaxNotificationFilterParametersCount is a reasonable filter's parameter limit
+// that does not allow attackers to increase node resources usage but that
+// also should be enough for real applications.
+const MaxNotificationFilterParametersCount = 16
```

Comment:
i dont mind but it is a const that is used exactly for a single purpose and defined right before the func that uses it. i would even do it as an internal const but thought that it would be useful to have some exported ref on what the requirements are